kubernetes.go

```diff
@@ -128,13 +130,79 @@ func homeDir() string {
 	return os.Getenv("USERPROFILE") // windows
 }
 
-func kubeGetReadinessHandler(ignoreDaemonSets bool) (readiness, error) {
+func kubeGetReadinessHandler(ignoreDaemonSets bool, deleteLocalData bool) (readiness, error) {
 	clientset, err := kubeGetClientset()
 	if err != nil {
 		log.Fatalf("Error getting kubernetes connection: %v", err)
 	}
 	if clientset == nil {
 		return nil, nil
 	}
-	return &kubernetesReadiness{clientset: clientset, ignoreDaemonSets: ignoreDaemonSets}, nil
+	return &kubernetesReadiness{clientset: clientset, ignoreDaemonSets: ignoreDaemonSets, deleteLocalData: deleteLocalData}, nil
+}
+
+// setScaleDownDisabledAnnotation set the "cluster-autoscaler.kubernetes.io/scale-down-disabled" annotation
+// on the list of nodes if required. Returns a list of hostname where the annotation
+// is applied.
+func setScaleDownDisabledAnnotation(hostnames []string) ([]string, error) {
+	// get the node reference - first need the hostname
+	var (
+		node      *corev1.Node
+		hostname  string
+		err       error
+		key       = "cluster-autoscaler.kubernetes.io/scale-down-disabled"
```

Comment:
Can we move this to a `const` at the beginning of the file?

Reply:
done